### ISSUE TITLE
Support for lazy loading of data

### ISSUE THREAD

**Issue by neilkuumar** [alonrbar/react-tablize#5]
Firstly, thank you for this library it works really well!

I am looking to implement lazy loading of data in conjunction with `react-tablize` (Grid) and just wanted to see if you had any ideas about how I would support it?

For example:
`GET /?page=1` returns 20 columns
user scrolls towards the end of the columns
`GET /?page=2` returns the next 20 columns
and so forth.

**Comment by alonrbar** [comment]
Hi, glad you enjoy it :)

Regarding lazy loading, as you already seen it's not currently supported. I was planning to add it once I gonna need that, although that did not happen yet. 

My plan was simple - expose the onScroll event and let the user of the library calculate whether the end of the list/table/graph was reached. He will then get the data necessary and update the rowCount property accordingly. This should generally work but I haven't tested this idea. I am however adding and removing rows and columns dynamically in my apps so I suppose this should work either.

In practice this means you'll have to modify this library a bit to expose the onScroll event (shouldn't be too hard to support) so I invite you to PR.

If you prefer not to modify the library, another approach that may work is to add some sort of "marker element" to be rendered by the list on a specific index and listen to it's onMount event to determine if new data should be added. This is pretty hacky though and I probably wouldn't have done it if I had the choice.

Let me know if what you think, I'll be happy to bounce some ideas.

**Comment by alonrbar** [comment]
Just released v2.1.3, thanks!
Note that the onScroll event has the following structre:
```typescript
export interface ScrollEvent {
    scrollTop: number;
    scrollLeft: number;
}
```

**Comment by alonrbar** [comment]
Closing the issue. Feel free to re-open if you feel anything is still missing.
